packages/location-management/src/components/LocationTree/index.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { ChangeEvent, useState } from 'react';
+import { Input, Tree as AntTree } from 'antd';
+import { SearchOutlined } from '@ant-design/icons';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import { ParsedHierarchyNode } from '../../ducks/types';
+import reducer, { reducerName } from '../../ducks/location-hierarchy';
+import { AntTreeProps } from '../LocationUnitView';
+import './tree.css';
+reducerRegistry.register(reducerName, reducer);
+
+interface TreeProp {
+  data: ParsedHierarchyNode[];
+  OnItemClick: (item: ParsedHierarchyNode) => void;
+}
+
+const Tree: React.FC<TreeProp> = (props: TreeProp) => {
+  const { data, OnItemClick } = props;
+
+  const [expandedKeys, setExpandedKeys] = useState<React.Key[]>([]);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [autoExpandParent, setAutoExpandParent] = useState<boolean>(true);
+  const filterData: ParsedHierarchyNode[] = [];
+
+  /** Return the the parent key in a tree for the supplied key
+   *
+   * @param {string} key the key to find parent of
+   * @param {Array<ParsedHierarchyNode>} tree the orignal tree
+   * @returns {string} - returns parent key
+   */
+  function getParentKey(key: string, tree: ParsedHierarchyNode[]): string {
+    let nodeKey = '';
+    tree.forEach((node) => {
+      if (node.children) {
+        if (node.children.some((item: ParsedHierarchyNode) => item.parent === key)) {
+          nodeKey = node.key;
+        } else if (getParentKey(key, node.children)) return getParentKey(key, node.children);
+      }
+    });
+    return nodeKey;
+  }
+
+  /** Function to handle event when a tree is expanded
+   *
+   * @param {Array<React.Key>} allExpandedKeys currently expanded keys
+   */
+  function onExpand(allExpandedKeys: React.Key[]) {
+    setExpandedKeys(allExpandedKeys);
+    setAutoExpandParent(true);
+  }
+
+  /** Function to handle event when tree serach input changes value
+   *
+   * @param {ChangeEvent<HTMLInputElement>} event the actual event
+   */
+  function onChange(event: ChangeEvent<HTMLInputElement>) {
+    const { value } = event.target;
+    const expandedKeys = filterData
+      .map((item) =>
+        value.length && item.label.toLocaleLowerCase().indexOf(value) > -1
+          ? getParentKey(item.id, filterData)
+          : null
+      )
+      .filter((item, i, self) => item && self.indexOf(item) === i);
+    setExpandedKeys(expandedKeys as string[]);
+    setSearchValue(value);
+    setAutoExpandParent(value.length > 0);
+  }
+
+  /** process the data before it could be displayed in tree
+   *
+   * @param {Array<ParsedHierarchyNode[]>} data the tree data to preprocess
+   * @returns {object} - returns obj with title, key and children
+   */
+  function loop(data: ParsedHierarchyNode[]): AntTreeProps[] {
```

Comment:
Nitpicky, Instead of loop, I think something in the lines of buildTreeData would be a better name

Reply:
done 👍